networking.adoc

```diff
@@ -0,0 +1,1081 @@
+
+[source,terminal]
+----
+$ oc adm create-bootstrap-project-template -o yaml > template.yaml
+----
+
+
+[source,terminal]
+----
+$ oc create -f template.yaml -n openshift-config
+----
+
+
+[source,terminal]
+----
+$ oc edit template <project_template> -n openshift-config
+----
+
+
+[source,terminal]
+----
+$ oc new-project <project> 
+----
+
+## Deleting a network policy
+
+[source,terminal]
+----
+$ oc delete networkpolicy <policy-name>
+----
+
+## Editing a network policy
+
+[source,terminal]
+----
+$ oc apply -f <policy-file>.yaml
+----
+
+
+[source,terminal]
+----
+$ oc edit <policy-name> -n <namespace>
+----
+
+## Configuring egress IPs for a project
+
+[source,terminal]
+----
+ $ oc patch netnamespace <project_name> --type=merge -p \ 
+  '{
+    "egressIPs": [
+      "<ip_address>" 
+    ]
+  }'
+----
+
+
+[source,terminal]
+----
+$ oc patch hostsubnet <node_name> --type=merge -p \ 
+  '{
+    "egressCIDRs": [
+      "<ip_address_range_1>", "<ip_address_range_2>" 
+    ]
+  }'
+----
+
+
+[source,terminal]
+----
+$ oc patch hostsubnet node1 --type=merge -p \
+  '{"egressCIDRs": ["192.168.1.0/24"]}'
+$ oc patch hostsubnet node2 --type=merge -p \
+  '{"egressCIDRs": ["192.168.1.0/24"]}'
+----
+
+
+[source,terminal]
+----
+$ oc patch netnamespace <project> --type=merge -p \ 
+  '{
+    "egressIPs": [ 
+      "<ip_address>"
+      ]
+  }'
+----
+
+
+[source,terminal]
+----
+$ oc patch hostsubnet <node_name> --type=merge -p \ 
+  '{
+    "egressIPs": [ 
+      "<ip_address_1>",
+      "<ip_address_N>"
+      ]
+  }'
+----
+
+
+[source,terminal]
+----
+$ oc patch hostsubnet node1 --type=merge -p \
+  '{"egressIPs": ["192.168.1.100", "192.168.1.101", "192.168.1.102"]}'
+----
+
+## Configuring an egress firewall for a project
+
+[source,terminal]
+----
+$ oc create -f <policy_name>.yaml -n <project>
+----
+
+## Configuring an egress router pod destination list from a config map
+
+[source,terminal]
+----
+$ oc delete configmap egress-routes --ignore-not-found
+----
+
+
+[source,terminal]
+----
+$ oc create configmap egress-routes \
+  --from-file=destination=my-egress-destination.txt
+----
+
+## Configuring kube-proxy
+
+[source,terminal]
+----
+$ oc edit network.operator.openshift.io cluster
+----
+
+## Deploying an egress router pod in DNS proxy mode
+
+[source,terminal]
+----
+$ oc create -f egress-router-service.yaml
+----
+
+## Disabling multicast for a project
+
+[source,terminal]
+----
+$ oc annotate netnamespace <namespace> \ 
+    netnamespace.network.openshift.io/multicast-enabled-
+----
+
+## Editing an egress firewall for a project
+
+[source,terminal]
+----
+$ oc replace -f <filename>.yaml
+----
+
+## Enabling multicast for a project
+
+[source,terminal]
+----
+$ oc annotate netnamespace <namespace> \
+    netnamespace.network.openshift.io/multicast-enabled=true
+----
+
+## Configuring network isolation using OpenShift SDN
+
+[source,terminal]
+----
+$ oc adm pod-network join-projects --to=<project1> <project2> <project3>
```

Comment:
Requires a declarative alternative

Reply:
@siamaksade, all the instances of `oc adm pod-network ...` are related to OpenShift SDN in multitenant mode.  This area of OpenShift is not in active development (so little that there is no automation to start a cluster in this mode) and developing updates for this area in the doc would be counter to the limited development of the software.

I don't plan to pursue updates to commands that are related to multitenant mode.